evals/features/cve/test_suggest_cwe.py

```diff
@@ -0,0 +1,89 @@
+import pytest
+
+from pydantic_evals import Case, Dataset
+from pydantic_evals.evaluators import Evaluator, EvaluatorContext
+
+from aegis.agents import rh_feature_agent
+from aegis.features.cve import SuggestCWE, SuggestCWEModel
+
+from evals.features.common import common_feature_evals, handle_eval_report
+
+
+class SuggestCweCase(Case):
+    def __init__(self, cve_id, cwe_list):
+        """cve_id given as CVE-YYYY-NUM is the flaw we query CWE for.  cwe_list
+        is the list of acceptable CWEs, the most preferred one comes first"""
+        super().__init__(
+            name=f"suggest-cwe-for-{cve_id}",
+            inputs=cve_id,
+            expected_output=cwe_list,
+            metadata={"difficulty": "easy"},
+        )
+
+
+class SuggestCweEvaluator(Evaluator[str, SuggestCWEModel]):
+    @staticmethod
+    def _base_score(cwe_list_out, cwe_list_exp):
+        score = 1.0
+        for cwe_exp in cwe_list_exp:
+            for cwe in cwe_list_out:
+                if cwe == cwe_exp:
+                    return score
+                score *= 0.9
+            score *= 0.9
+
+        # no match
+        return 0.0
+
+    def evaluate(self, ctx: EvaluatorContext[str, SuggestCWEModel]) -> float:
+        """return score based on actual and expected results"""
+        cwe_list_out = ctx.output.cwe
+        score = self._base_score(cwe_list_out, ctx.expected_output)
+
+        # check how many CWEs were suggested and how man CWEs are accepted
+        len_diff = len(cwe_list_out) - len(ctx.expected_output)
+        if 0 < len_diff:
+            # penalize too many suggested CWEs for a CVE
+            score *= 0.9**len_diff
+
+        conf_diff = ctx.output.confidence - score
+        if 0.0 < conf_diff:
+            # penalize confident models providing (partially) wrong results
+            score -= conf_diff
+        else:
+            # negligibly penalize models providing correct results but low confidence
+            score += conf_diff / 4.0
```

Comment:
might be worth extruding 4.0 into a constant ... not sure there is value in making constants env vars as well eg. once we settle on their values they should not change often.